### DISCUSSION TITLE
Is there a way to change `[]*Thing` to `[]Thing`?

### DISCUSSION THREAD

**Discussion by frederikhors** [ent/ent#2613] [Q&A] [answered]
Using `gqlgen` I'm using `[]Thing` everywhere instead of `[]*Thing` with the option `omit_slice_element_pointers` in config file `gqlgen.yml`.

Can I ask why you preferred to use this type of signature with pointers?

Is there a way to change it like in `gqlgen`?

**Comment by a8m** [comment, marked as answer]
If you mean for an entity (eager-loaded) edges that are a few reasons for this, but the main 2 are: this is how Ent distinguishes between Found/NotFound (see [example](https://github.com/ent/ent/blob/master/examples/start/ent/car.go#L47-L51)) and in some cases, we reuse the same object in a few references. 

For `gqlgen` integration, we also return pointers to indicate `null` (not found) object. Here's an example: https://github.com/ent/contrib/blob/master/entgql/internal/todo/ent/gql_edge.go#L164
